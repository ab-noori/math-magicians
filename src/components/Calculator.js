// import React, { useState } from 'react';

const Calculator = () => (
  <div className="calculator">
    <div className="display">0</div>
    <div className="row">
      <button type="button" className="btn clear">AC</button>
      <button type="button" className="btn plus-minus">+/-</button>
      <button type="button" className="btn percent">%</button>
      <button type="button" className="btn operator">&divide;</button>
    </div>
    <div className="row">
      <button type="button" className="btn number">7</button>
      <button type="button" className="btn number">8</button>
      <button type="button" className="btn number">9</button>
      <button type="button" className="btn operator">&times;</button>
    </div>
    <div className="row">
      <button type="button" className="btn number">4</button>
      <button type="button" className="btn number">5</button>
      <button type="button" className="btn number">6</button>
      <button type="button" className="btn operator">&minus;</button>
    </div>
    <div className="row">
      <button type="button" className="btn number">1</button>
      <button type="button" className="btn number">2</button>
      <button type="button" className="btn number">3</button>
      <button type="button" className="btn operator">+</button>
    </div>
    <div className="row">
      <button type="button" className="btn zero number">0</button>
      <button type="button" className="btn decimal">.</button>
      <button type="button" className="btn equals operator">=</button>
    </div>
  </div>
);

export default Calculator;
