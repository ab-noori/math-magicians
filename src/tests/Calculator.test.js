import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('Calculator component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
