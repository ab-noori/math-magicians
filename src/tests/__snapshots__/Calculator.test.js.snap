// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders correctly 1`] = `
<div
  className="calculator"
>
  <div
    className="display"
  >
    0
  </div>
  <div
    className="row"
  >
    <button
      className="btn clear"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="btn plus-minus"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="btn percent"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn operator"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="btn number"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn number"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn number"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn operator"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="btn number"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn number"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn number"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn operator"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="btn number"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn number"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn number"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn operator"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="btn number zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn decimal"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="btn equals operator"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
